### PR TITLE
Add experimental redis support

### PR DESCRIPTION
**Pull request by peterschwarz**
This PR adds a Redis-backed database implementation of the Database trait.  It is available behind an experimental feature flag, `"redis-db"`.

The set of tests added to the state integration tests cover the standard operations, as well as verifying that the same hashes are produced when a B-Tree-backed database is used.

This implementation does not currently support the DatabaseCursor-related methods.

Additionally, this PR moves the Merkle-Radix state tests to the integration test path, and applies them across all current `Database` implementations.

To test locally, either run `./bin/run_tests` or

```
$ docker run -p 6379:6379 --name some-redis -d redis
$ cargo test --manifest-path libtransact/Cargo.toml redis --features redis-db
```

### CODE REVIEW

File: libtransact/tests/state.rs [hunk 1]
```diff
@@ -14,558 +14,534 @@
...
+        }
 
-            assert_eq!(orig_root_bytes, &change_log.parent);
-            assert_eq!(3, change_log.additions.len());
-            assert_eq!(0, change_log.successors.len());
+        let state_change = StateChange::Set {
+            key: "abcd".to_string(),
+            value: "data_value".as_bytes().to_vec(),
+        };
+        //merkle_db.set_merkle_root(orig_root.clone()).unwrap();
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libtransact/tests/state.rs]: That was in the original test moved there, but will remove in an additional commit.

File: libtransact/tests/state.rs [hunk 2]
```diff
@@ -1303,10 +1303,11 @@ mod state_tests {
             let redis_url =
                 std::env::var("REDIS_URL").unwrap_or_else(|_| DEFAULT_REDIS_URL.to_string());
             let mut rng = thread_rng();
-            let db_name: String = iter::repeat(())
-                .map(|()| rng.sample(Alphanumeric))
-                .take(7)
-                .collect();
+            let db_name = String::from("test-db-")
```
[the remark this answers is not in the document]
Comment (peterschwarz) [reply libtransact/tests/state.rs]: Added it to the wrong commit

File: libtransact/src/database/mod.rs [hunk 3]
```diff
@@ -33,7 +33,7 @@ pub mod btree;
 pub mod error;
 pub mod lmdb;
 
-use crate::database::error::DatabaseError;
```
Comment (agunde406) [inline libtransact/src/database/mod.rs]: typo in commit message: integegration -> integration

Comment (peterschwarz) [reply libtransact/src/database/mod.rs]: Fixed

**Pull request merged**